### PR TITLE
fixed error in rank-regulators.rkt

### PR DESCRIPTION
**Pull request by mzheng17**

### CODE REVIEW

File: biolink/pieces-parts/rank-regulators.rkt [hunk 1]
```diff
@@ -1,8 +1,11 @@
 #lang racket
 (require "query.rkt"
+         "rank-regulators-gene-lists.rkt"
          racket/hash)
 (provide (all-defined-out)
-         (all-from-out "query.rkt" racket/hash))
+         (all-from-out "query.rkt"
+                       "rank-regulators-gene-lists.rkt"
```
Comment (namin) [inline biolink/pieces-parts/rank-regulators.rkt:7]: why this change?

File: biolink/pieces-parts/rank-regulators.rkt [hunk 2]
```diff
@@ -315,11 +318,11 @@ function) because that edge is the one that has the directional predicates.
...
                                (lambda (v)
                                  (cond
-                                   [(assoc (car new-val) v) => (lambda (a) 
-                                                                 (unless (empty? pubmeds)
-                                                                   (set-add (set-remove v a)
-                                                                            (list (car a)
-                                                                                  (set-union (cadr a) pubmeds)))))]
+                                   [(and (not (empty? pubmeds))
+                                         (assoc (car new-val) v)) => (lambda (a)
+                                                                       (set-add (set-remove v a)
```
Comment (namin) [inline biolink/pieces-parts/rank-regulators.rkt:323]: nice!

File: biolink/pieces-parts/rank-regulators.rkt [hunk 3]
```diff
@@ -315,11 +318,11 @@ function) because that edge is the one that has the directional predicates.
                                (concept->curie/name g)
                                (lambda (v)
                                  (cond
-                                   [(assoc (car new-val) v) => (lambda (a) 
-                                                                 (unless (empty? pubmeds)
-                                                                   (set-add (set-remove v a)
-                                                                            (list (car a)
-                                                                                  (set-union (cadr a) pubmeds)))))]
+                                   [(and (not (empty? pubmeds))
```
Comment (namin) [inline biolink/pieces-parts/rank-regulators.rkt:321]: What's the problem with an empty pubmeds?

**Review by namin — COMMENTED** [review]
Just a few questions left.

Thanks for this change.

File: biolink/pieces-parts/rank-regulators.rkt (hunk 1 above)
Comment (mzheng17) [reply biolink/pieces-parts/rank-regulators.rkt:7]: I just included that so the user can just load rank-regulators.rkt and not have to load rank-regulators-gene-lists.rkt separately. With my workflow, it was just easier this way.

File: biolink/pieces-parts/rank-regulators.rkt (hunk 3 above)
Comment (mzheng17) [reply biolink/pieces-parts/rank-regulators.rkt:321]: All the inside of that cond statement does is add the pubmed ids in the variable pubmeds to the provenance that's being recorded, so it would be a waste to execute the inside statements if pubmeds is empty.

Comment (namin) [reply biolink/pieces-parts/rank-regulators.rkt:321]: But is it the right thing to do the else case here if the pubmed is empty?

Comment (mzheng17) [reply biolink/pieces-parts/rank-regulators.rkt:321]: Oh, I see. I think the way I would try to fix that tho is to write what is in the deleted lines from the last commit, but those caused an error, so I'll look into how to fix it.

Comment (namin) [reply biolink/pieces-parts/rank-regulators.rkt:321]: I think the simplest way is to not treat pubmeds empty so specially, maybe only save the set-union within the first (now not pubmed specific) branch. Does it make sense?

Comment (namin) [reply biolink/pieces-parts/rank-regulators.rkt:321]: But I think you can rely on racket not being wasteful in that case, so I would just remove the pubmed logic entirely.

Comment (mzheng17) [reply biolink/pieces-parts/rank-regulators.rkt:321]: Yeah that makes sense.

File: biolink/pieces-parts/rank-regulators.rkt (hunk 1 above)
Comment (namin) [reply biolink/pieces-parts/rank-regulators.rkt:7]: Do you think it might make more sense to have a rank-regulators-lib file which has both rank-regulators and rank-regulators-gene-lists imported, then import that one for the workflow?

I did run into the same workflow need for what it's worth.

Or maybe add a comment that this is not used here, but just more convenient?

Anyways, I think it's OK like that. :)
